packages/components/src/Auth.tsx

```diff
@@ -289,32 +292,41 @@ function LoginButton({ mnemonic, chain, network, path, url }: any) {
 
 function LoginForm() {
   const [mnemonic, setMnemonic] = useState<string>(new Computer().getMnemonic())
-  const [chain, setChain] = useState<Chain | undefined>(process.env['REACT_APP_CHAIN'] as Chain | undefined)
-  const [network, setNetwork] = useState<Network | undefined>(process.env['REACT_APP_NETWORK'] as Network | undefined)
-  const [url, setUrl] = useState<string | undefined>(process.env['REACT_APP_URL'])
+  const [chain, setChain] = useState<Chain | undefined>(
+    process.env["REACT_APP_CHAIN"] as Chain | undefined
+  )
+  const [network, setNetwork] = useState<Network | undefined>(
+    process.env["REACT_APP_NETWORK"] as Network | undefined
+  )
+  const [url, _] = useState<string | undefined>(process.env["REACT_APP_URL"])
+  const urlInputRef = useRef<HTMLInputElement>(null)
 
   useEffect(() => {
     initFlowbite()
-  }, [])
+  }, [urlInputRef.current])
 
   return (
     <>
       <div className="max-w-sm mx-auto p-4 md:p-5 space-y-4">
         <form className="space-y-6">
           <div>
             <MnemonicInput mnemonic={mnemonic} setMnemonic={setMnemonic} />
-            {!process.env['REACT_APP_CHAIN'] && <ChainInput chain={chain} setChain={setChain} />}
-            {!process.env['REACT_APP_NETWORK'] && <NetworkInput network={network} setNetwork={setNetwork} />}
-            {!process.env['REACT_APP_URL'] && <UrlInput setUrl={setUrl} />}
+            {!process.env["REACT_APP_CHAIN"] && <ChainInput chain={chain} setChain={setChain} />}
+            {!process.env["REACT_APP_NETWORK"] && (
+              <NetworkInput network={network} setNetwork={setNetwork} />
+            )}
+            {!process.env["REACT_APP_URL"] && <UrlInput urlInputRef={urlInputRef} />}
```

Comment:
Maybe we should pass url and setUrl in here. Unless that is what causes slowness.